src/libraries/Microsoft.Extensions.Logging.Abstractions/tests/Microsoft.Extensions.Logging.Generators.Tests/LoggerMessageGeneratorParserTests.cs

```diff
@@ -80,6 +80,23 @@ partial class C
             Assert.Equal(DiagnosticDescriptors.LoggingMethodHasBody.Id, diagnostics[0].Id);
         }
 
+        [Fact]
+        public async Task InvalidMethodExpressionBody()
+        {
+            IReadOnlyList<Diagnostic> diagnostics = await RunGenerator(@"
+                partial class C
+                {
+                    static partial void M1(ILogger logger);
+
+                    [LoggerMessage(EventId = 0, Level = LogLevel.Debug, Message = ""M1"")]
+                    static partial void M1(ILogger logger) => throw new Exception();
```

Comment:
Sounds like a very reasonable idea. Just to clarify, the existing method body test does not cover this test case either. Would be good to add for both `Body` and `ExpressionBody` tests?

Reply:
Yes - I think it would be ideal for both to be tested.

Reply:
What's the expected behavior for the following code? 
```
[LoggerMessage(EventId = 0, Level = LogLevel.Debug, Message = ""M1"")]
static partial void M1(ILogger logger);

static partial void M1(ILogger logger) { }
```

Currently, this code won't generate any diagnostic because the implementation doesn't have the `LoggerMessageAttribute` applied and all the checks are skipped. Source code is [here](https://github.com/dotnet/runtime/blob/9028dc2d715a827a21a5ffbd029bc7a24073406b/src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs#L102).

If the user does this in their solution, they are likely to get a [CS0756 compile error](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0756) because the partial method `M1` has more than one implementation.

It's probably a separate and a bigger issue than the OG bug we're trying to fix just IMHO.

Reply:
> IMHO It's probably a separate and a bigger issue than the OG bug we're trying to fix, and I'm happy to create a separate issue.

I agree this is a separate (probably lower priority) issue. If you can create it separately, that would be great.

Reply:
Issue created. Feel free to triage and close it if you think it's low priority and affecting less.